### ISSUE TITLE
After upgrade getting storage_proxy - Failed to apply mutation from x std::out_of_range (deserialization buffer underflow)

### ISSUE THREAD

**Issue by mrkkhattak** [scylladb/scylla#4816]
I upgraded from 3.0.6-0.20190503.bf68fae01 to 3.1.0.rc3-0.20190725.bebfd7b26 on one of our node.  

This is our staging cluster consisting of 2 nodes, hosted on AWS and using open source AWS AMI for Scylla. 

After upgrade, I started scylla and now I see these messages in logs:

> ....scylla.bin[27801]:  [shard 1] storage_proxy - Failed to apply mutation from x.x.x.x#0: std::out_of_range (deserialization buffer underflow)

I upgraded by following https://docs.scylladb.com/upgrade/upgrade-opensource/upgrade-guide-from-3.0-to-3.1/upgrade-guide-from-3.0-to-3.1-rpm

I have been upgrading from 2.x till now (3-4 times) on same cluster without any issue but now seeing this error. 

This might be related to but as not sure so opening a new issue:
https://github.com/scylladb/scylla/issues/3100
https://github.com/scylladb/scylla/issues/4132

**Comment by gleb-cloudius** [comment]
Do you see those errors constantly after upgrade or they appeared only
while of the nodes were been stopped? Can you provide more logs? Can you
see any logs about connection been closed near those errors?

--
			Gleb.

**Comment by slivne** [comment]
@mrkkhattak it will greatly help us if you can add information

**Comment by mrkkhattak** [comment]
> Do you see those errors constantly after upgrade or they appeared only while of the nodes were been stopped? Can you provide more logs? Can you see any logs about connection been closed near those errors?
> […](#)
> -- Gleb.

I continuously see this error appearing in our node logs. Due to this error I didn't upgrade our second node.

**Comment by mrkkhattak** [comment]
> @mrkkhattak it will greatly help us if you can add information

I see mostly this line appearing in logs when viewed via _journalctl _COMM=scylla_

Apart from this, as I already mentioned it is a test cluster of 2 nodes hosted on AWS. Nodes are very small but have been running without any issue from last 15+ months. Load is very minimal on this cluster.

If there is anything else required, please let me know.

**Comment by gleb-cloudius** [comment]
On Thu, Aug 08, 2019 at 05:49:31AM -0700, Meraj Rasool Khattak wrote:
> If there is anything else required, please let me know.
> 
The log may still help. Also your schema and queries.

--
			Gleb.

**Comment by johsbk** [comment]
Observing the same behaviour after upgrading from 3.0.10 to 3.1.0 on two out of three instances of the cluster, it happens continously:
Oct 21 08:10:32 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 2] storage_proxy - Failed to apply mutation from 10.52.5.209#2: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:32 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 5] storage_proxy - Failed to apply mutation from 10.52.5.209#5: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:32 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 4] storage_proxy - Failed to apply mutation from 10.52.5.209#4: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:32 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 5] storage_proxy - Failed to apply mutation from 10.52.3.143#5: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:32 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 0] storage_proxy - Failed to apply mutation from 10.52.5.209#0: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:32 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 1] storage_proxy - Failed to apply mutation from 10.52.3.143#1: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:32 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 3] storage_proxy - Failed to apply mutation from 10.52.3.143#3: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:32 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 2] storage_proxy - Failed to apply mutation from 10.52.3.143#2: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 1] storage_proxy - Failed to apply mutation from 10.52.5.209#1: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 3] storage_proxy - Failed to apply mutation from 10.52.5.209#3: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 2] storage_proxy - Failed to apply mutation from 10.52.5.209#2: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 5] storage_proxy - Failed to apply mutation from 10.52.5.209#5: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 4] storage_proxy - Failed to apply mutation from 10.52.5.209#4: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 5] storage_proxy - Failed to apply mutation from 10.52.3.143#5: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 0] storage_proxy - Failed to apply mutation from 10.52.5.209#0: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 1] storage_proxy - Failed to apply mutation from 10.52.3.143#1: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 3] storage_proxy - Failed to apply mutation from 10.52.3.143#3: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:33 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 2] storage_proxy - Failed to apply mutation from 10.52.3.143#2: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:34 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 1] storage_proxy - Failed to apply mutation from 10.52.5.209#1: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:34 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 3] storage_proxy - Failed to apply mutation from 10.52.5.209#3: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:34 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 2] storage_proxy - Failed to apply mutation from 10.52.5.209#2: std::out_of_range (deserialization buffer underflow)
Oct 21 08:10:34 ip-10-52-2-24.eu-central-1.compute.internal scylla[22890]:  [shard 5] storage_proxy - Failed to apply mutation from 10.52.5.209#5: std::out_of_range (deserialization buffer underflow)

**Comment by tgrabiec** [comment]
Duplicate of  #4855

**Comment by tgrabiec** [comment]
@johsbk I recommend rolling back to 3.0.10 for now, removing commitlog and hinted handoff files on the rolled-back nodes before starting the 3.0.10 version.

**Comment by johsbk** [comment]
@tgrabiec to follow up - the error was observed the most at the first node - I guess because of the incompatible objects mentioned in #4855 was found in commitlogs / hints written by 3.0.10 on the other nodes - the issue was solved by clearing commitlogs and hints on the second and third nodes. As such I'm able to continue using 3.1.0
